lms/static/scripts/frontend_apps/utils/dashboard/hooks.ts

```diff
@@ -0,0 +1,54 @@
+import { useCallback, useMemo } from 'preact/hooks';
+import { useLocation, useSearch } from 'wouter-preact';
+
+import { queryStringToRecord, recordToQueryString } from '../url';
+
+export type DashboardFilters = {
+  studentIds: string[];
+  assignmentIds: string[];
+  courseIds: string[];
+};
+
+export type UseDashboardQueryFilters = {
+  filters: DashboardFilters;
+  updateFilters: (filtersToUpdate: Partial<DashboardFilters>) => void;
+};
+
+/**
+ * Allows to track dashboard filters as a piece of state synced in the query string
+ */
+export function useDashboardQueryFilters(): UseDashboardQueryFilters {
```

Comment:
I think we can simplify the name here to `useDashboardFilters`, since we don't need to disambiguate with any other kind of filter. Same with the associated types.

Reply:
I thought it would help making it more clear that the query params are involved when using this, and is not just a wrapper around a bunch of `useState`s, but I agree the name is not great, so I'll change it for now and we can re-evaluate later if we find any problem or confusion.